### ISSUE TITLE
Promise wrapper, cb is not a function, and connection not released

### ISSUE THREAD

**Issue by magnusjt** [sidorares/node-mysql2#509]
Using the promise wrapper and pool.execute without any args gives an error "callback is not a function" when the result of the query is an error. In addition, if the result is not an error, the connection is not released.

In the promise wrapper we have (createPool):

````js
execute: function (sql, args) {
  return new Promise(function (resolve, reject) {
	var done = makeDoneCb(resolve, reject);
	if (args) {
	  corePool.execute(sql, args, done);
	} else {
	  corePool.execute(sql, done);
	}
  });
}
````

The second line `corePool.execute(sql, done);` is called when we don't supply any args.

In pool.execute we have:

````js
Pool.prototype.execute = function (sql, values, cb) {
  var useNamedPlaceholders = this.config.connectionConfig.namedPlaceholders;

  this.getConnection(function (err, conn) {
    if (err) {
      return cb(err);
    }

    conn.config.namedPlaceholders = useNamedPlaceholders;
    return conn.execute(sql, values, function () {
      conn.release();
      cb.apply(this, arguments);
    });
  });
};
````

Since cb is undefined here, if there is an error we get "cb is not a function". If there isn't an error, conn.execute is run, but the supplied function is not run. Instead, "values" is taken as the callback function (which translates to the done callback in the promise wrapper), but then `conn.release()` is never run.

**Comment by sushantdhiman** [comment]
I think its fixed with https://github.com/sidorares/node-mysql2/pull/488 , not released to npm registry yet!!

**Comment by sidorares** [comment]
@magnusjt I'll try to prepare release in next 12 hours

**Comment by magnusjt** [comment]
Ok, good :) Thanks for the quick response!

Another thing I should point out is that the promise-wrapper as it stands now will swallow any exceptions thrown by the wrapped functions (such as the error thrown here). I don't know if this is anything to worry about though.

**Comment by sidorares** [comment]
> Another thing I should point out is that the promise-wrapper as it stands now will swallow any exceptions thrown by the wrapped functions (such as the error thrown here). I don't know if this is anything to worry about though.

good point, I might wrap in try/catch and reject promise on exception

**Comment by sidorares** [comment]
@magnusjt I just published v1.2.0, please reopen if you still have the issue
